consumer/consumer.go

```diff
@@ -156,9 +156,10 @@ func (c consumerGroupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, cla
 	// and it's not possible to remove a topic from the handler list.
 	h := c.consumer.handlers.get(claim.Topic())
 	for msg := range claim.Messages() {
+		mark := true
```

Comment:
Just to understand this better, would this mean that if there is an error and we don't  have `config.Discarded` defined we will be committing the message? 
Should we actually not commit it if there is an error so it gets back in the queue? Or maybe only when there is context error we set the mark = false, so at least we don't skip messages on context error 🤔?

Sorry if I got this wrong somehow 🙈

Reply:
> Just to understand this better, would this mean that if there is an error and we don't have `config.Discarded` defined we will be committing the message?

Yes. This is how library works from the beginning :see_no_evil: so I'm keeping backwards compatibility.

> Should we actually not commit it if there is an error so it gets back in the queue? Or maybe only when there is context error we set the mark = false, so at least we don't skip messages on context error thinking?

On context error from ConsumerGroupSession, the messages are *NOT* committed, so I think will not call `Discarded` in that case. It makes no sense

> Sorry if I got this wrong somehow see_no_evil

Not at all, better to keep it clear.

Reply:
> On context error from ConsumerGroupSession, the messages are NOT committed, so I think will not call Discarded in that case. It makes no sense

It actually will call it if defined, we are in the error block, or I look this differently :). 

Although, you say the messages are NOT committed and wondering if this is the code 
```
for range claim.Messages() {}
```
that puts the messages back in the queue and the following command on line174 `sess.MarkMessage(msg, "")` won't have an affect on marking the message since we will reach here because we set the mark default as true and in case if we don't have a `config.Discarded` defined.

Reply:
Added check in 4a43622b8ce46f0e39ca72777f58934f05001084 to avoid confusion.